src/robot/libdocpkg/jsonbuilder.py

```diff
@@ -0,0 +1,81 @@
+#  Copyright 2020-     Robot Framework Foundation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import json
+import os.path
+
+from robot.errors import DataError
+from robot.running.arguments import ArgumentSpec, ArgInfo
+
+from .model import LibraryDoc, KeywordDoc
+
+
+class JsonDocBuilder(object):
+
+    def build(self, path):
+        spec = self._parse_spec_json(path)
+        return self.build_from_dict(spec)
+
+    def build_from_dict(self, libdoc_dict):
+        libdoc = LibraryDoc(name=libdoc_dict.get('name'),
+                            doc=libdoc_dict.get('doc'),
+                            version=libdoc_dict.get('version'),
+                            type=libdoc_dict.get('type'),
+                            scope=libdoc_dict.get('scope'),
+                            doc_format=libdoc_dict.get('doc_format'),
+                            source=libdoc_dict.get('source'),
+                            lineno=int(libdoc_dict.get('lineno')))
+        libdoc.inits = self._create_keywords(libdoc_dict.get('inits'))
+        libdoc.keywords = self._create_keywords(libdoc_dict.get('keywords'))
+        return libdoc
+
+    @staticmethod
+    def _parse_spec_json(path):
+        if not os.path.isfile(path):
+            raise DataError("Spec file '%s' does not exist." % path)
+        with open(path) as json_source:
+            libdoc_dict = json.load(json_source)
+        return libdoc_dict
+
+    def _create_keywords(self, keywords):
+        return [KeywordDoc(name=kw.get('name'),
+                           args=self._create_arguments(kw.get('args')),
+                           doc=kw.get('doc'),
+                           shortdoc=kw.get('shortdoc'),
+                           tags=kw.get('tags'),
+                           source=kw.get('source'),
+                           lineno=int(kw.get('lineno'))) for kw in keywords]
```

Comment:
- Unfortunate that args need special handling and this cannot changed just to `KeywordDoc(**kw)`.

- Not all keywords have line number information so `int()` conversion may sometimes fail. If line number is stored in the spec as an integer or null, no conversion is needed.